### ISSUE TITLE
Getting incorrect version mismatch when deploying functions app 

### ISSUE THREAD

**Issue by mattstewart1189** [Azure/Azure-Functions#921]
I'm trying to deploy a V2 functions app. The deployment is configured to use github.

The deployment failed with the following message:

_**Your function app is targeting V1, but Azure host has function version V2, \r\nplease change the version using the portal or update your 'FUNCTIONS_EXTENSION_VERSION' appsetting and retry**_

The portal is set to V2 
![runtimeversion](https://user-images.githubusercontent.com/1912580/44645599-c7b09c00-a99d-11e8-9f90-1eb46947de8c.png)

![inkedappsetting_li](https://user-images.githubusercontent.com/1912580/44645604-cd0de680-a99d-11e8-963b-7f107628f963.jpg)

My functions app is set to V2 as well:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <AzureFunctionsVersion>v2</AzureFunctionsVersion>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="autofac" Version="4.8.1" />
    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="1.0.3" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.14" />
    <PackageReference Include="Twilio.AspNet.Core" Version="5.9.7" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\DraftDay.Model\DraftDay.Model.csproj" />
    <ProjectReference Include="..\DraftDay.DataAccess\DraftDay.DataAccess.csproj" />
    <ProjectReference Include="..\DraftDay.Services\DraftDay.Services.csproj" />
    <ProjectReference Include="..\DraftDay.Shared\DraftDay.Shared.csproj" />
  </ItemGroup>
</Project>
```

**Comment by paulbatum** [comment]
@mattstewart1189 can you clarify exactly how you are deploying? You mentioned github, but I'm not sure what component would be showing that error in the case of continuous deployment through github. That error can be shown in VS.. are you using VS?

**Comment by mattstewart1189** [comment]
@paulbatum sorry, yes.

In azure, I configured the deployment options to connect to a branch in a github repo. It's the first time that I had done it, but after I push a commit the app gets redeployed.

I'm using VS Code, to develop. The error was in the log in the Azure portal. Under deployment options --> Deployments --> Deployment Details.

My functions app was apart of a solution. The entire solution and all associated projects were checked into the same GitHub repository. I guess when the deployment executed it may not have been able to find the functions app?

After I submitted this issue, I disconnected the GitHub deployment, removed the functions app from my solution and deployed the functions app through VS code via the Azure Functions extension and the deployment was successful.

I assumed the GitHub deploy could track down the functions app. I would like my functions app to be able to belong to the solution so that all of the dependencies get compiled from source at deploy time.

**Comment by paulbatum** [comment]
I just found out there was a bug that might have been causing this behavior, and we have deployed a fix last week. If you try again you might find it works as expected now.

However I should note that we do not have infrastructure that "finds" the function app in your solution. There is a configuration setting you can use to tell the infrastructure which project to deploy:
https://github.com/projectkudu/kudu/wiki/Customizing-deployments#deploying-a-specific-aspnet-or-aspnet-core-project-file

So you might need to use that setting to point to the project for the function app.

cc @davidebbo

**Comment by mattstewart1189** [comment]
Thanks, @paulbatum. This is certainly a step in the right direction. I've added the app setting and the project is being correctly targeted.

Now the build is failing during deployment with some sort of issue with assemblies relating to the Microsoft.Net.Sdk.Functions package. The application compiles locally. I've attached the deploy log.

[FailedDeployLog.txt](https://github.com/Azure/Azure-Functions/files/2326639/FailedDeployLog.txt)
